### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by ruimonteiro93**
Initial implementation of react-lib-template.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,8 @@
 # react-lib-template
-This template aims to make the implementation of react component packages easier and more methodic.
+A template that aims to make the implementation of React component packages easier and more methodic.
+
+## Usage
+
+This repo serves as a template for whenever you want to create a new package with an extractable React component. For this, just select `@moxystudio/react-lib-template` it as the template when creating a new repo for your new package, and you're all set to start working on your new package.
```
Comment (paulobmarcos) [inline README.md:6]: Suggestion:
This repo serves as a template to the creation of moxy's base React components.

If you want to use this, just select `@moxystudio/react-lib-template` as the template while creating a new repo for your new package and you're all set to start working.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,8 @@
...
+
+This template already includes the `src` folder, with 2 dummy files ready for you to start your work. `NewComponent` was chosen as the dummy name, just rename `NewComponent.js` according to your needs. An `index.js` for exporting is available as well. Do not forget to add unit tests for each.
```
Comment (paulobmarcos) [inline README.md:8]: Suggestion:

"
This template already includes a `src` folder, with 2 dummy files to guide you on where to start. `NewComponent`, a dummy component, is available for demonstration purposes but you can change it according to your needs.

Also, do not forget to add unit tests and to try and reach maximum coverage.

If you want to use this, just select @moxystudio/react-lib-template as the template while creating a new repo for your new package and you're all set to start working.
"

File: package.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "@moxy/react-lib-template",
+  "version": "0.0.0",
+  "description": "An orchestrator that eases out the implementation of page transitions",
```
Comment (guilhermedeandrade) [inline package.json:4]: This was probably the description from the project you used as reference, right?

File: package.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  "main": "lib/index.js",
+  "module": "es/index.js",
+  "files": [
+    "lib",
+    "es"
+  ],
+  "homepage": "https://github.com/moxystudio/react-lib-template#readme",
+  "author": "Rui Monteiro <rui.monteiro@moxy.studio>",
+  "license": "MIT",
+  "keywords": [
```
Comment (guilhermedeandrade) [inline package.json:14]: My comment above can be applied here as well

File: package.json (hunk 3 above)
Comment (ruimonteiro93) [reply package.json:4]: Eheh, yes 😄 I will correct it, thanks.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1 @@
+// Use this to export the Component and any other related utilities you are about to create
```
Comment (fsdiogo) [inline src/index.js:1]: I'd argue that we should add this code and change the text to instruct just to rename the component.

```js
export { default } from './NewComponent';
```

File: src/NewComponent.js [hunk 6]
```diff
@@ -0,0 +1 @@
+// Here, create the new Component you want to see exported to a new package
```
Comment (fsdiogo) [inline src/NewComponent.js:1]: Following my other comment, I'd like to have a sample here, e.g.:

```js
import React from 'react';
import PropTypes from 'prop-types';

const NewComponent = ({ children }) => (
    <div>
        { children }
    </div>
);

NewComponent.propTypes = {
    children: PropTypes.any.isRequired,
};

export default NewComponent;
```

Comment (ruimonteiro93) [reply src/NewComponent.js:1]: Creating a sample is an excellent idea, and the addition of prop-types has value as well. But should we go with receiving children? Wouldn't it be better to render a simple dummy text? We would add another prop just for example, non-required perhaps.

Comment (fsdiogo) [reply src/NewComponent.js:1]: I've added `children` just to have an excuse to add the `PropTypes` really, but any will do!

A dummy text LGTM 👍

**Comment by fsdiogo** [comment]
I would also add a `NewComponent.test.js` file with a dummy test.

File: src/NewComponent.js (hunk 6 above)
Comment (ruimonteiro93) [reply src/NewComponent.js:1]: Actually, now that I think of it, adding a children prop makes perfect sense here. It's a common use case and it serves as a good "excuse" to add PropTypes. Let's go with your suggestion 😸

File: package.json (hunk 3 above)
Comment (paulobmarcos) [reply package.json:4]: Maybe I would add something here that reminds people to change it 🤔 Similarly to what we have in `next-with-moxy`

**Comment by paulobmarcos** [comment]
Maybe we should add a checklist to the README to change stuff like:

- [ ] Change `package.json` name, description, keywords, etc...
- [ ] Add tests
- [ ] Update the README
- [ ] ...

What do you guys think?

**Comment by PedroMiguelSS** [comment]
Yap, I think that would be helpful @paulobmarcos. Having a mini-recipe to guide people on how to create a new component and make sure nothing is left behind.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,15 @@
...
+
+## Usage
+
+This repo serves as the template for the creation of MOXY's base `React` components. To use this, just select `@moxystudio/react-lib-template` as the template when creating a new repo for your new package, and you're all set to start working.
+
+This template already includes a `src` folder, with 2 dummy files ready for you to start your work. `NewComponent` is a dummy component, available for demonstration purposes. Just rename `NewComponent.js` and change it according to your needs. An `index.js` for exporting is available as well. Do not forget to update the unit tests and try to reach as much coverage as possible.
+
+In order to help make proper use of this template, here's a quick checklist with some crucial stuff to have in mind:
+
+- Remeber to change `package.json` name, description, keywords, etc.
```
Comment (fsdiogo) [inline README.md:12]:
```suggestion
- Remember to change `package.json` name, description, keywords, etc.
```

File: src/NewComponent.test.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import NewComponent from './NewComponent';
+
+it('should render no layout', () => {
+    const wrapper = render(
+        <NewComponent>
+            <span>Hello</span>
+        </NewComponent>,
+    );
+
+    expect(wrapper).toMatchSnapshot();
```
Comment (PedroMiguelSS) [inline src/NewComponent.test.js:12]: What do you think of changing this test to something like:

```js
const { getByText } = render(
     <NewComponent>
          <span>Hello</span>
     </NewComponent>,
);

expect(getByText('Hello')).toBeInTheDocument();
```

This way, we would avoid the usage of snapshot.
